metrics/fidelity/fidelity.py

```diff
@@ -0,0 +1,145 @@
+import logging
```

Comment:
Overall comments:
Looks pretty good, but I think you should add 
1) type hinting
2) more specific, imperative function names
and pick a consistent terminology for rationales. You call them "rationales", "alpha" and "attention_masks" at various points in the code.

See specific comments for more details